src/backend/app/data-source/repository-post.js

```diff
@@ -36,4 +39,130 @@ const createPost = async (posts) => {
   }
 }
 
-module.exports = { getPostById, createPost };
+const updatePostIndexStatus = async (indexHistory) => {
+  try {
+    return await post.update({
+      index_status: indexHistory.status
+    }, {
+      where: {
+        id: indexHistory.post_id
+      }
+    });
+  } catch (err) {
+    console.log("Error : Could not update Post status");
+    throw err;
+  }
+}
+
+const datasourceIndexStatus = async () => {
+  try {
+    const data = await post.findAll({
```

Comment:
Can you checkout the API for findAndCountAll ? ([link](https://sequelize.org/master/class/lib/model.js~Model.html#static-method-findAndCountAll)) 
that will give you the total count for response. which might be useful to convert absolute values into percentages.

Not a high priority task. ok to ignore.